src/topology/algebra/module/finite_dimension.lean

```diff
@@ -0,0 +1,356 @@
+/-
+Copyright (c) 2022 Anatole Dedecker. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Sébastien Gouëzel, Anatole Dedecker
+-/
+import analysis.locally_convex.balanced_core_hull
+
+/-!
+# Finite dimensional topological vector spaces over complete fields
+
+Over a complete nondiscrete field, in finite dimension, all norms are equivalent and all linear maps
+are continuous.
+
+## Main results :
+
+* `linear_map.continuous_iff_is_closed_ker` : a linear form is continuous if and only if its kernel
+  is closed.
+* `linear_map.continuous_of_finite_dimensional` : a linear map on a finite-dimensional space over a
+  complete field is continuous.
+
+## TODO
+
+Generalize more of `analysis/normed_space/finite_dimension` to general TVSs.
+
+## Implementation detail
+
+The main result from which everything follows is the fact that, if `ξ : ι → E` is a finite basis,
+then `ξ.equiv_fun : E →ₗ (ι → 𝕜)` is continuous. However, for technical reasons, it is easier to
+prove this when `ι` and `E` live ine the same universe. So we start by doing that as a private
+lemma, then we deduce `linear_map.continuous_of_finite_dimensional` from it, and then the general
+result follows as `continuous_equiv_fun_basis`.
+
+-/
+
+universes u v w x
+
+noncomputable theory
+
+open set finite_dimensional topological_space filter
+open_locale classical big_operators filter topological_space nnreal uniformity
```

Comment:
I went with option 1 so I had to fix things in a few other files. Do you want me to split that into a separate PR or is it okay ?

> I wonder is there something more general to prove here?

I don't see how it could be made really more general because it's just a combination of three arguments : 
- `f` can be decomposed in the canonical basis (general fact about the module structure on `ι → 𝕜`)
- each function `x ↦ x i` is continuous (general fact about the product topology)
- any linear combination of continuous functions `* → F` is continuous because `F` is a TVS)
And these are the interesting general facts, so I guess the "general version" is already there, just in the form of those three lemmas.